src/main/java/com/google/step/datamanager/TagManagerDatastore.java

```diff
@@ -72,4 +74,17 @@ private Tag transformEntityToTag(Entity entity) {
 
     return new Tag(id, name);
   }
+
+  @Override
+  public List<Tag> readTags(List<Long> ids) {
+    List<Tag> tags = new ArrayList<>();
+    for (long id : ids) {
+      try {
+        tags.add(readTag(id));
+      } catch (IllegalArgumentException e) {
+        continue;
+      }
+    }
+    return tags;
+  }
```

Comment:
Is this the final implementation? Can we add tests for this?

Reply:
ok. I refactored read users and read tags according to comments in #57 and add tests.